azure-application-insights-spring-boot-starter/src/main/java/com/microsoft/applicationinsights/autoconfigure/conditionals/InstrumentationKeyCondition.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * ApplicationInsights-Java
+ * Copyright (c) Microsoft Corporation
+ * All rights reserved.
+ *
+ * MIT License
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the ""Software""), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+package com.microsoft.applicationinsights.autoconfigure.conditionals;
+
+import com.microsoft.applicationinsights.autoconfigure.helpers.IkeyResolver;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.context.annotation.Condition;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+
+/**
+ * Conditional to check if instrumentation key is either set in application.properties
+ * or as system property or environment variable.
+ *
+ * @author Dhaval Doshi
+ */
+public class InstrumentationKeyCondition implements Condition {
+
+  @Override
+  public boolean matches(ConditionContext context, AnnotatedTypeMetadata metadata) {
```

Comment:
`@ConditionalOnProperty` already searches in those location. Is there a reason why you can't use it?

Reply:
@snicoll due to historic reasons users of Application could set instrumentation key with three different names either in environment variables or properties which is : `APPLICATION_INSIGHTS_IKEY`and `APPINSIGHTS_INSTRUMENTATIONKEY` therefore I needed to verify if the instrumentation key is present as either of the three and therefore I had to come up with this custom conditional instead of `@ConditionalOnProperty` annotation.

Reply:
You should extends from `SpringBootCondition` then because your custom condition doesn't show up in the auto-config report. I also find the doc a bit misleading: it should describe the fact the keys are different.

Reply:
Sounds good. Thanks for the suggestion here, and I will improve the comments to make it more relevant.